language/stdlib/modules/libra_system_2.move

```diff
@@ -0,0 +1,264 @@
+    public fun get_validator_info(addr: &address): ValidatorInfo acquires ValidatorSet {
+        let validator_set_ref = borrow_global<ValidatorSet>(0x1D8);
+        Transaction::assert(is_validator_(addr, &validator_set_ref.validators), 19);
+
+        let i = get_validator_index_(&validator_set_ref.validators, addr);
+        let validator_info = Vector::borrow(&validator_set_ref.validators, i);
+        return *validator_info
+    }
+
+    // Return the size of the current validator set
+    public fun validator_set_size(): u64 acquires ValidatorSet {
+        Vector::length(&borrow_global<ValidatorSet>(0x1D8).validators)
+    }
+
+    // Private helper functions
+
+    // Emit a reconfiguration event. This function will be invoked by the genesis to generate the very first
+    // reconfiguration event.
+    fun emit_reconfiguration_event() acquires ValidatorSet {
+        let validator_set_ref = borrow_global_mut<ValidatorSet>(0x1D8);
+
+        // Ensure only one reconfiguration event is emitted per transaction (or per block? TODO(valerini))
+        let current_block_time = LibraTimestamp::now_microseconds();
+        
+        // Abort the transaction if reconfiguration was already called within the same block
+        Transaction::assert(current_block_time > validator_set_ref.last_reconfiguration_time, 14);
+        
+        validator_set_ref.last_reconfiguration_time = current_block_time;
+        LibraAccount::emit_event<ValidatorSetChangeEvent>(
+            &mut validator_set_ref.change_events,
+            ValidatorSetChangeEvent {
+                new_validator_set: *&validator_set_ref.validators,
+            });
+    }
+
+    // Get the index of the validator with address `addr` in `validators`.
+    // Aborts if `addr` is not a current validator
+    fun get_validator_index_(validators: &vector<ValidatorInfo>, addr: &address): u64 {
+        let len = Vector::length(validators);
+        let i = 0;
+        loop {
+            if (get_validator_address(Vector::borrow(validators, i)) == addr) {
+                return i
+            };
+
+            i = i + 1;
+            if (i >= len) { break };
+        };
+
+        abort 99
+    }
+
+    // Updates ith validator info, if nothing changed, return false
+    fun update_ith_validator_info_(validators: &mut vector<ValidatorInfo>, i: u64): bool {
+        let size = Vector::length(validators);
+        if (i >= size) {
+            return false
+        };
+        let validator_info = Vector::borrow(validators, i);
+        let new_validator_config = ValidatorConfig2::get_config(validator_info.addr);
+        // check if information is the same
+        if (&Vector::borrow(validators, i).config == &new_validator_config) {
+            return false
+        };
+        *&mut Vector::borrow_mut(validators, i).config = *&new_validator_config;
+        return true
+    }
+
+    fun is_validator_(addr: &address, validators_vec_ref: &vector<ValidatorInfo>): bool {
```

Comment:
seems you could share code between get_validator_index_ and this so you don't have duplicate code

Reply:
That's great, fixing.